main.py

```
@@ -181,12 +181,35 @@ def zip_sb3(fnc, project_dir):
+# GUI FUNCTIONS
+# To connect to the Tkinter window
+class DownloadController:
+    def __init__(self):
+        self.session = None
+    def validate_login(self, username, password):
+        if password:
+            try:
+                self.session = s3.login(username, password)
+                print("Login successful!")
+                return True
+            except Exception as e:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)